### PR TITLE
Threading

### PR DESCRIPTION
**Pull request by jonay2000**
Fixes: #18

### CODE REVIEW

**Comment by jonay2000** [comment]
@vgel Hey there! I think I merged it correctly now. Your drop tests pass :). I Added the same MaybeUninit code to the AllocRingbuffer because when the buffer gets full it may need to replace items with uninitialized items. Thanks again for the help!

**Comment by jonay2000** [comment]
It's now possible to pre-initialize the AllocRingbuffer with Maybeuninit items. That might make the AllocRingBuffer a little faster.

File: Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "ringbuffer"
-version = "0.4.0"
+version = "1.1.0"
```
Comment (NULLx76) [inline Cargo.toml:3]: 0.4.0 will be the next release not 1.1.0

File: benches/bench.rs [hunk 2]
```diff
@@ -79,7 +79,7 @@ macro_rules! generate_benches {
 fn criterion_benchmark(c: &mut Criterion) {
     c.with_plots();
 
-    generate_benches![
+    generat_benches![
```
Comment (NULLx76) [inline benches/bench.rs:82]: typo, should be `generate_benches`

File: src/lib.rs [hunk 3]
```diff
@@ -74,6 +76,11 @@ pub use with_alloc::AllocRingBuffer;
 #[cfg(feature = "alloc")]
 pub use with_alloc::RINGBUFFER_DEFAULT_CAPACITY;
 
+// #[cfg(all(feature = "alloc", feature = "threads"))]
+// mod with_threads_alloc;
+// #[cfg(all(feature = "alloc", feature = "threads"))]
+// pub use with_threads_alloc::ThreadAllocRingBuffer;
```
Comment (NULLx76) [inline src/lib.rs:82]: remove or uncomment

File: src/with_threads_alloc.rs [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+/// The AllocRingBuffer is a RingBuffer which is based on a Vec. This means it allocates at runtime
+/// on the heap, and therefore needs the [`alloc`] crate. This struct and therefore the dependency on
+/// alloc can be disabled by disabling the `alloc` (default) feature.
+///
+/// # Example
+/// ```rust
+/// // TODO: Example
+/// ```
+#[derive(PartialEq, Eq, Debug)]
+pub struct ThreadAllocRingBuffer<T> {
```
Comment (NULLx76) [inline src/with_threads_alloc.rs:17]: I think we want to add this in the future so for now I suggest we branch from this branch, remove the threaded code an make a new PR?

**Pull request closed without merging**